pynio/service.py

```diff
@@ -53,7 +53,12 @@ def connect(self, blk1, blk2=None):
         # if block exists, add the receiever, otherwise init connection
         receivers = [] if blk2 is None else [blk2.name]
         if connection:
-            connection['receivers'].extend(receivers)
+            cr = connection['receivers']
+            cr.extend(receivers)
+            # remove duplicates but preserve order
+            seen = set()
+            seen_add = seen.add
```

Comment:
really we should probably add the `json()` method and use a dictionary of sets for connections anyway. This would get around this whole issue and would make it easier to interface with in the future (such as renaming blocks across an instance)